tests/test_yaml_search_space/test_search_space.py

```diff
@@ -0,0 +1,159 @@
+    pipeline_space = pipeline_space_from_yaml(
+        "tests/test_yaml_search_space/correct_config_including_priors.yml"
+    )
+    assert isinstance(pipeline_space, dict)
+    assert isinstance(pipeline_space["learning_rate"], FloatParameter)
+    assert pipeline_space["learning_rate"].lower == 0.00001
+    assert pipeline_space["learning_rate"].upper == 0.1
+    assert pipeline_space["learning_rate"].log is True
+    assert pipeline_space["learning_rate"].is_fidelity is False
+    assert pipeline_space["learning_rate"].default == 3.3e-2
+    assert pipeline_space["learning_rate"].default_confidence_score == 0.125
+    assert isinstance(pipeline_space["num_epochs"], IntegerParameter)
+    assert pipeline_space["num_epochs"].lower == 3
+    assert pipeline_space["num_epochs"].upper == 30
+    assert pipeline_space["num_epochs"].log is False
+    assert pipeline_space["num_epochs"].is_fidelity is True
+    assert pipeline_space["num_epochs"].default == 10
+    assert pipeline_space["num_epochs"].default_confidence_score == 0.5
+    assert isinstance(pipeline_space["optimizer"], CategoricalParameter)
+    assert pipeline_space["optimizer"].choices == ["adam", 90e-3, "rmsprop"]
+    assert pipeline_space["optimizer"].is_fidelity is False
+    assert pipeline_space["optimizer"].default == 90e-3
+    assert pipeline_space["optimizer"].default_confidence_score == 4
+    assert isinstance(pipeline_space["dropout_rate"], ConstantParameter)
+    assert pipeline_space["dropout_rate"].value == 1e3
+    assert pipeline_space["dropout_rate"].default == 1e3
+
+
+@pytest.mark.neps_api
+def test_incorrect_yaml_file():
+    """Test the function with an incorrectly formatted YAML file."""
+    with pytest.raises(SearchSpaceFromYamlFileError) as excinfo:
+        pipeline_space_from_yaml(
+            Path("tests/test_yaml_search_space/incorrect_config.txt")
+        )
+    assert str(excinfo.value.exception_type == "ValueError")
+
+
+@pytest.mark.neps_api
+def test_yaml_file_with_missing_key():
+    """Test the function with a YAML file missing a required key."""
+    with pytest.raises(SearchSpaceFromYamlFileError) as excinfo:
+        pipeline_space_from_yaml("tests/test_yaml_search_space/missing_key_config.yml")
+    assert str(excinfo.value.exception_type == "KeyError")
+
+
+@pytest.mark.neps_api
+def test_yaml_file_with_inconsistent_types():
+    """Test the function with a YAML file having inconsistent types for
+    'lower' and 'upper'."""
+    with pytest.raises(SearchSpaceFromYamlFileError) as excinfo:
+        pipeline_space_from_yaml(
+            "tests/test_yaml_search_space/inconsistent_types_config.yml"
+        )
+    assert str(excinfo.value.exception_type == "TypeError")
+    with pytest.raises(SearchSpaceFromYamlFileError) as excinfo:
+        pipeline_space_from_yaml(
+            Path("tests/test_yaml_search_space/inconsistent_types_config2.yml")
+        )
+    assert str(excinfo.value.exception_type == "TypeError")
```

Comment:
Since a boolean expression is converted to either "True" or "False" the `assertion` will always be True.
Are we sure these test cases pass? 
We should check these before releasing them?

Reply:
You are right. im going to fix that.

Reply:
Should be fixed[here](https://github.com/automl/neps/pull/46/commits/075c79b67944a881b6153d2f1e79ac4633ef73ae)